src/feature/PdkCommandFeature.ts

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+import * as vscode from 'vscode';
+import { IFeature } from '../feature';
+import { ILogger } from '../logging';
+import { PDKCommandStrings } from '../messages';
+import { reporter } from '../telemetry/telemetry';
+
+export class PdkCommandFeature implements IFeature {
+  private logger: ILogger;
+  private terminal: vscode.Terminal;
+
+  constructor(context:vscode.ExtensionContext, logger: ILogger) {
+    this.logger = logger;
+    this.terminal = vscode.window.createTerminal('Puppet PDK');
+    this.terminal.processId.then(pid => {
+      logger.debug('pdk shell started, pid: ' + pid);
+    });
+    context.subscriptions.push(this.terminal);
+
+    [
+      PDKCommandStrings.PdkNewModuleCommandId,
+      PDKCommandStrings.PdkNewClassCommandId,
+      PDKCommandStrings.PdkNewTaskCommandId,
+      PDKCommandStrings.PdkTestUnitCommandId,
+      PDKCommandStrings.PdkValidateCommandId
+    ].forEach((id)=>{
+      context.subscriptions.push(vscode.commands.registerCommand(id, () => {
+        this.pdkNewModule();
```

Comment:
Not sure if this was your intent.  Every command will call this.pdkNewModule();  I assume what you really mean is

``` typescript
context.subscriptions.push(vscode.commands.registerCommand(PDKCommandStrings.PdkNewModuleCommandId, () => {
        this.pdkNewModule();
}
this.logger.debug("Registered " + PDKCommandStrings.PdkNewModuleCommandId+ " command");

context.subscriptions.push(vscode.commands.registerCommand(PDKCommandStrings.PdkNewClassCommandId, () => {
        this.pdkNewClass();
}
this.logger.debug("Registered " + PDKCommandStrings.PdkNewClassCommandId+ " command");

...
```